src/context/EarnManagerProvider/components/Withdraw/Withdraw.tsx

```diff
@@ -1,28 +1,389 @@
+              >
+                {/* This should display the opposite field */}
+                {cryptoField ? (
+                  <Amount.Fiat value={values?.fiatAmount || ''} />
+                ) : (
+                  <Amount.Crypto value={values?.cryptoAmount || ''} symbol={asset.symbol} />
+                )}
+              </FormHelperText>
+            </Box>
+            <VStack
+              bg={useColorModeValue('gray.50', 'gray.850')}
+              borderRadius='xl'
+              borderWidth={1}
+              borderColor={useColorModeValue('gray.100', 'gray.750')}
+              divider={<Divider />}
+              spacing={0}
+            >
+              <InputGroup size='lg'>
+                <InputLeftElement ml={1} width='auto'>
+                  <Button
+                    ml={1}
+                    size='sm'
+                    variant='ghost'
+                    textTransform='uppercase'
+                    onClick={handleInputToggle}
+                    width='full'
+                  >
+                    {cryptoField ? asset.symbol : 'USD'}
+                  </Button>
+                </InputLeftElement>
+                {cryptoField && (
+                  <Controller
+                    render={({ field: { onChange, value } }) => {
+                      return (
+                        <NumberFormat
+                          customInput={CryptoInput}
+                          decimalSeparator={localeParts.decimal}
+                          inputMode='decimal'
+                          thousandSeparator={localeParts.group}
+                          value={value}
+                          onChange={e => {
+                            onChange(amountRef.current)
+                            handleInputChange(amountRef.current as string)
+                            amountRef.current = null
+                          }}
+                          onValueChange={e => {
+                            amountRef.current = e.value
+                          }}
+                        />
+                      )
+                    }}
+                    name={Field.CryptoAmount}
+                    control={control}
+                    rules={cryptoInputValidation}
+                  />
+                )}
+                {!cryptoField && (
+                  <Controller
+                    render={({ field: { onChange, value } }) => {
+                      return (
+                        <NumberFormat
+                          customInput={CryptoInput}
+                          decimalSeparator={localeParts.decimal}
+                          inputMode='decimal'
+                          thousandSeparator={localeParts.group}
+                          value={bnOrZero(value).toFixed(2)}
+                          onChange={e => {
+                            onChange(amountRef.current)
+                            handleInputChange(amountRef.current as string)
```

Comment:
We should try to avoid `as` if there is another way to check the type.
```suggestion
                            if (amountRef.current) handleInputChange(amountRef.current)
```